pkg/utils/net/ip.go

```diff
@@ -6,12 +6,45 @@ package net
 
 import (
 	"net"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
 )
 
-// MaybeIPTo4 attempts to convert the provided net.IP to a 4-byte representation if possible, otherwise does nothing.
-func MaybeIPTo4(ipAddress net.IP) net.IP {
-	if ip := ipAddress.To4(); ip != nil {
+// IPToRFCForm normalizes the IP address given to fit the expected network byte order octet form described in
+// https://tools.ietf.org/html/rfc5280#section-4.2.1.6
```

Comment:
IIUC in addition to the byte ordering requirement we also want to avoid compressed zero form of IPv6 addresses using "::" ?
Maybe we can also explain this in the comment ?

Reply:
Well it is not removing the compressed form actually. Which is fine , with the exception of Enterprise Search which has currently an issue with compressed IPv6 IPs. I missed this because I was testing with manually configured overrides 😞 So thanks for pointing this out, will fix in another commit.